[FLINK-13057][state] Correct comments in ListState class

hequn8128 opened the pull request:
## What is the purpose of the change

This pull request fixes the comments in `ListState`. 

`ListState` can be a keyed state or an operator state, but the comment in `ListState` said it can only be a keyed state


## Brief change log

  - Correct comments in `ListState` class, i.e. `ListState` can be a keyed state or an operator state.


## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

klion26 commented on flink-core/src/main/java/org/apache/flink/api/common/state/ListState.java:
```diff
@@ -27,10 +27,14 @@
- * key of the current element. That way, the system can handle stream and state partitioning
- * consistently together.
+ * <p>The state can be a keyed list state or an operator list state.
+ * When it is a keyed list state, it is accessed by functions applied on a {@code KeyedStream}.
```
we can add a `<p>` before `When it is a keyed list state` and `When it is an operator list state` to get better format?

hequn8128 replied:
good suggestion!

zentol commented:
CI report for commit a0cd89a69dbce15adc26d5dc28ef49640ab8c8c8: FAILURE [Travis](https://travis-ci.org/dataArtisans/flink/builds/554657500?utm_source=github_status&utm_medium=notification)

hequn8128 commented:
> CI report for commit [a0cd89a](https://github.com/apache/flink/commit/a0cd89a69dbce15adc26d5dc28ef49640ab8c8c8): FAILURE [Travis](https://travis-ci.org/dataArtisans/flink/builds/554657500?utm_source=github_status&utm_medium=notification)

@zentol Thanks a lot for the review. 
As for the FAILURE, it seems some dependencies could not be resolved. Maybe it is not related to this PR? or something I have missed? 

I have updated my PR and rebased to the latest master. I will check travis and see whether the problem still exists. Thanks a lot.

Best, 
Hequn

StephanEwen commented:
Good fix, thank you!

Merging this...

hequn8128 commented:
@StephanEwen Thank you very much. :-)
